DeleteButton.js

```diff
@@ -10,7 +10,7 @@ export class DeleteButton {
     const deleteUserResponse = await fetch(`${this.url}/${this.userId}`, {
       method: "DELETE",
     });
-    if (this.userContainer) {
+    if (this.userContainer && deleteUserResponse) {
```

Comment:
`deleteUserResponse` will always be truthy. You are not checking the actual response status, please follow the documentation to find solution https://developer.mozilla.org/en-US/docs/Web/API/Response/ok